src/mpi/mpi_caf.c

```diff
@@ -2842,492 +2798,463 @@ PREFIX(send) (caf_token_t token, size_t offset, int image_index,
-                }
-            }
+            convert_type(dest->base_addr + dst_offset, dst_type,
+                         dst_kind, sr, src_type, src_kind, stat);
+        }
+        else
+        {
+          dprint("%d/%d: %s() strided same_image, *WITH* temp, for i = %d.\n",
+                 caf_this_image, caf_num_images, __FUNCTION__, i);
+          if (same_type_and_kind)
+            memmove(t_buff + i * dst_size, sr, src_size);
+          else
+            convert_type(t_buff + i * dst_size, dst_type, dst_kind,
+                         sr, src_type, src_kind, stat);
+        }
+      }
 
 #ifndef WITH_FAILED_IMAGES
-          if (ierr != 0)
-            {
-              caf_runtime_error("MPI Error: %d", ierr);
-              return;
-            }
+      if (ierr != 0)
+      {
+        caf_runtime_error("MPI Error: %d", ierr);
+        return;
+      }
 #endif
-        }
-      if (!same_image)
-        CAF_Win_unlock(remote_image, *p);
-
+    }
+    if (!same_image)
+      CAF_Win_unlock(remote_image, *p);
 
-      if (same_image && mrt)
+    if (same_image && mrt)
+    {
+      for (i = 0; i < size; ++i)
+      {
+        ptrdiff_t array_offset_dst = 0, extent = 1, tot_ext = 1;
+        if (dst_vector == NULL)
         {
-          for (i = 0; i < size; ++i)
-            {
-              ptrdiff_t array_offset_dst = 0;
-              ptrdiff_t extent = 1;
-              ptrdiff_t tot_ext = 1;
-              if (dst_vector == NULL)
-                {
-                  for (j = 0; j < dst_rank - 1; ++j)
-                    {
-                      extent =
-                        dest->dim[j]._ubound - dest->dim[j].lower_bound + 1;
-                      array_offset_dst +=
-                        ((i / tot_ext) % extent) * dest->dim[j]._stride;
-                      tot_ext *= extent;
-                    }
+          for (j = 0; j < dst_rank - 1; ++j)
+          {
+            extent = dest->dim[j]._ubound - dest->dim[j].lower_bound + 1;
+            array_offset_dst += ((i / tot_ext) % extent) * dest->dim[j]._stride;
+            tot_ext *= extent;
+          }
 
-                  array_offset_dst +=
-                    (i / tot_ext) * dest->dim[dst_rank - 1]._stride;
-                }
-              else
-                {
-                  switch (dst_vector->u.v.kind)
-                    {
-                      // KINDCASE is defined above.
-                      KINDCASE(1, int8_t);
-                      KINDCASE(2, int16_t);
-                      KINDCASE(4, int32_t);
-                      KINDCASE(8, int64_t);
+          array_offset_dst += (i / tot_ext) * dest->dim[dst_rank - 1]._stride;
+        }
+        else
+        {
+          switch (dst_vector->u.v.kind)
+          {
+            // KINDCASE is defined above.
+            KINDCASE(1, int8_t);
+            KINDCASE(2, int16_t);
+            KINDCASE(4, int32_t);
+            KINDCASE(8, int64_t);
```

Comment:
Debug builds in [this commit](https://github.com/sourceryinstitute/OpenCoarrays/pull/555/commits/55ed8c01a1f79b79e6d97523ae2ec481b5ee77ed) fail to compile due to [errors attributed to lines 3240-3243](https://travis-ci.org/sourceryinstitute/OpenCoarrays/jobs/395903748#L1865-L1877) with [all versions of GCC/GFortran we are testing](https://travis-ci.org/sourceryinstitute/OpenCoarrays/builds/395903746):

```
/home/travis/build/sourceryinstitute/OpenCoarrays/src/mpi/mpi_caf.c:3240:25: error: expected expression before ‘int8_t’
             KINDCASE(1, int8_t);
                         ^~~~~~
# etc.
```

Something in this commit broke all the builds. I appreciate the cleanup, and would welcome it if you can resolve the issue, but I would also be content with resolving this with a `git reset HEAD^ ; git push --force`

Reply:
Sorry, pushed too soon, fixing this issue right now.